torch/csrc/jit/script/compiler.cpp

```diff
@@ -1308,44 +1308,27 @@ struct to_ir {
 
   void emitLoopCommon(
       SourceRange range,
-      c10::optional<Expr> max_trip_count,
-      c10::optional<Expr> cond,
       const List<Stmt>& body,
-      c10::optional<Ident> itr_ident,
-      bool in_list = false) {
+      std::function<void(Value*, std::shared_ptr<Environment>)> current_element_assigner,
+      c10::optional<Expr> cond,
+      Value* max_trip_count_val = nullptr
+      ) {
+    Value* cond_val = nullptr;
     Node* n = graph->insertNode(create(prim::Loop, range, 0));
-    Value *max_trip_count_val, *cond_val;
+    WithInsertPoint guard(n);
+
+    if (!max_trip_count_val)
     {
-      WithInsertPoint guard(n);
-      if (max_trip_count) {
-        if (in_list) {
-          auto listArg = emitExpr(max_trip_count.value());
-
-          max_trip_count_val = emitBuiltinCall(
-              max_trip_count->range(),
-              *graph,
-              aten::len,
-              c10::nullopt,
-              {listArg},
-              {},
-              /*required=*/true);
-        } else {
-          max_trip_count_val = ensureInt(
-              max_trip_count->range(), emitExpr(max_trip_count.value()));
-        }
-      } else {
-        max_trip_count_val = materializeConstant(
-            std::numeric_limits<int64_t>::max(),
-            *graph,
-            range,
-            integral_constants);
-      }
-      if (cond) {
-        cond_val = emitCond(cond.value());
-      } else {
-        cond_val = graph->insertConstant(true, nullptr, range);
-      }
+      max_trip_count_val = materializeConstant(
+          std::numeric_limits<int64_t>::max(),
+          *graph,
+          range,
+          integral_constants);
     }
+
+    auto true_constant  = graph->insertConstant(true, nullptr, range);
```

Comment:
We don't always want to insert the constant into the graph if a condition is provided right? This should be inside an `if`

Reply:
@driazati I'm counting on DCE to get rid of it when we use user's condition. I think having `true_constant` helps readability a little bit, but I can definitely remove `true_constant` and insert it only when needed. How strongly do you feel about this?

Reply:
You can keep `true_constant` and just remove the ternary so it only gets inserted in the else branch, I don't think coupling this to a pass (even one pretty much guaranteed to run like DCE) is a great precedent to set